src/afrolabs/components/kafka/bytes_serdes.clj

```diff
@@ -37,13 +38,22 @@
            :main false
            :implements [org.apache.kafka.common.serialization.Deserializer])
 
-(defn deser-deserialize
-  ([_ _ byte-data]
+(defn deser-deserialize-String-Headers-byte<>
+  ([_ _ ^bytes byte-data]
    (when byte-data
      (java.util.Arrays/copyOf ^bytes byte-data
                               (alength ^bytes byte-data))))
+  ([this _ _ ^bytes byte-data]
+   (deser-deserialize-String-Headers-byte<> this nil byte-data)))
+
+(defn deser-deserialize-String-Headers-ByteBuffer
+  ([_ _ ^ByteBuffer byte-data]
+   (when byte-data
+     (let [result (make-array Byte/TYPE (.remaining byte-data))]
+       (.get byte-data ^bytes result)
+       result)))
```

Comment:
The kafka deserializer interface unexpectedly gained a new overload on `deserialize`. This is a backwards compatible change for other JVM languages, but clojure has a bug where we are forced to implement this overload on `ByteBuffer`. More info here: https://pb.co.za/overloading-gen-class-methods-of-glorious-types-with-sublime-nation-of-clojure-for-make-great-benefit-of-interop